lab_09_experiments.py

```diff
@@ -131,37 +127,67 @@ def __init__(self):
         self.menu = Menu()
 
     def setup(self):
-        NUMBER_OF_WORMS = 10
+        NUMBER_OF_WORMS = 100
+        self.worms_killed = 0
+        self.in_base = True
+        self.player_speed = 300
+        self.speed_cost = 10
+        self.worm_receipt = "Not Enough Worms!!!"
+        self.show_worm_receipt = False
 
         self.scene = arcade.Scene()
 
-        self.player_sprite = Player("Mole.png", 0.25, center_x=500, center_y=100)
+        self.camera = arcade.Camera(1000, 1000)
+
+        self.base = arcade.Sprite("Base.png", 0.75, center_x=500, center_y=500)
+        self.scene.add_sprite("Base", self.base)
+
+        self.player_sprite = Player("Mole.png", 0.25, center_x=500, center_y=500)
         self.scene.add_sprite("Player", self.player_sprite)
 
         self.attack = arcade.load_animated_gif("Attack.gif")
 
+        self.worm_counter = Worm_Counter("Worm.png", 0.25)
+        self.scene.add_sprite("Counter", self.worm_counter)
+
         self.worms = arcade.SpriteList()
         for i in range(NUMBER_OF_WORMS):
             self.worms.append(Worm())
 
         self.scene.add_sprite_list(f"Worms", sprite_list=self.worms)
 
     def on_draw(self):
+        self.camera.use()
         self.clear()
         self.scene.draw()
         self.menu.draw()
+        message = ":" + str(self.worms_killed)
+        arcade.draw_text(message, self.player_sprite.center_x - 435, self.player_sprite.center_y + 422, arcade.color.BLACK, 30)
+
+        if self.in_base:
+            arcade.draw_text("Upgrade cost: " + str(self.speed_cost), 375, 350, arcade.color.BLACK, 25)
+            if self.show_worm_receipt:
+                arcade.draw_text(self.worm_receipt, 600, 500, arcade.color.BLACK, 30)
```

Comment:
The `on_draw` should be minimalistic with no logic inside!
`self.score.draw()`
`self.base?upgrade?.draw()`